src/request_builder.rs

```diff
@@ -243,12 +244,13 @@ impl Future for RequestBuilder {
     fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
         if self.fut.is_none() {
             let req = self.req.take().unwrap();
-            if let Some(client) = &self.client {
-                self.fut = Some(client.send(req))
-            } else {
-                let client = Client::new_shared_or_panic();
-                self.fut = Some(client.send(req))
-            }
+
+            let client = self
+                .client
+                .take()
```

Comment:
does this have the same behavior? previously we didn't consume the client

Reply:
`await` is consuming. I can't find a good reference, but it definitely is.

Reply:
Also see line 245/246 above: we already `let req = self.req.take().unwrap();`.